packages/doenetml-worker-rust/lib-doenetml-core/tests/by_component/xref.rs

```diff
@@ -49,3 +45,60 @@ fn xref_referring_to_itself_wont_crash_the_system() {
     let pv: PropView<prop_type::ComponentRef> = prop.into_prop_view();
     assert_eq!(pv.value.unwrap(), ComponentRef(xref_idx));
 }
+
+#[test]
+fn xref_ref_can_make_display_text() {
+    let dast_root = dast_root_no_position(r#"<section name="foo"/><xref ref="$foo" />"#);
+
+    let mut core = TestCore::new();
+    core.init_from_dast_root(&dast_root);
+
+    // the document tag will be index 0.
+    let xref_idx = ComponentIdx::from(2);
+
+    let prop = core.get_prop(xref_idx, XrefProps::DisplayText.local_idx());
+    let pv: PropView<prop_type::String> = prop.into_prop_view();
+    assert_eq!(*pv.value, "Section 1".to_string());
+}
+
+#[test]
+fn xref_ref_can_make_display_text_with_custom_children() {
+    let dast_root = dast_root_no_position(r#"<section name="foo"/><xref ref="$foo">Foo</xref>"#);
```

Comment:
From this test, I started out confused what the `Foo` string child is supposed to do. Maybe showing that it is the rendered child will at least make it seem like it wasn't lost?

Reply:
In the UI this should render as `Foo 1`, with a space between the `Foo` and the `1`